### DISCUSSION TITLE
Is there anyone who understands what is required?

### DISCUSSION THREAD

**Discussion by samirovic-7** [gradescope/autograder_samples#54] [Q&A] [answered]
UX and UI: Create 5 coding assignments which would teach students to develop their abilities to create good user interfaces and experiences.

**Comment by TheMapotakes** [comment]
Yes

**Comment by ibrahima** [comment, marked as answer]
Hi! Are you asking how to go about creating such assignments? Autograders are most helpful in cases where the "correctness" of a solution can be assessed programmatically. Gradescope's autograder platform is quite flexible so there's a lot that you can do as long as you can code it, but it may be difficult to write a program to determine whether something is a "good UI/UX" since that's a bit of an abstract concept. However, at its core, if you can write code that follows these [specifications](https://gradescope-autograders.readthedocs.io/en/latest/specs/), it can be an autograder on Gradescope.

For this case, it may make sense to write a series of assignments that introduces students to the tools and components they would use to build user interfaces. If students are asked to implement a specific design and specific functionality, it may be possible to automate testing that their application meets the expected requirements (for example, clicking on a button labeled "Sign up" should open up a sign up form with particular fields). For web development, you can use automation frameworks such as [Selenium](https://github.com/SeleniumHQ/selenium) or [capybara](https://github.com/teamcapybara/capybara/) to automate such tests. There may be similar testing tools for other UI programming frameworks. In general, the more specific you can make your assignment, the more likely it is to be autogradable.

I would also add that Gradescope does [have robust manual grading capabilities](https://gradescope-autograders.readthedocs.io/en/latest/manual_grading/), so if you wanted to start with a series of more rigid assignments to build familiarity with tools, and then give students the opportunity to get creative with an open-ended assignment, you could handle submission collection and grading on Gradescope, even if the assignment was not fully autogradable.

Hope that helps!
